wgengine/netstack/netstack.go

```diff
@@ -775,6 +787,13 @@ func (ns *Impl) forwardUDP(client *gonet.UDPConn, wq *waiter.Queue, clientAddr,
 		backendRemoteAddr = &net.UDPAddr{IP: net.ParseIP("127.0.0.1"), Port: int(port)}
 		backendListenAddr = &net.UDPAddr{IP: net.ParseIP("127.0.0.1"), Port: int(srcPort)}
 	} else {
+		if dstIP := dstAddr.IP(); viaRange.Contains(dstIP) {
+			dial16 := dstIP.As16()
+			dstAddr = netaddr.IPPortFrom(
+				netaddr.IPv4(dial16[12], dial16[13], dial16[14], dial16[15]),
```

Comment:
`netaddr.IPFrom4(dial16[12:])`?

Reply:
You mean `netaddr.IPFrom4(*(*[4]byte)(dial16[12:]))`? 🤮 😝

Reply:
But https://github.com/golang/go/issues/46505 would make it slightly less gross.

Reply:
Done. PTAL. Moved to a common func in tsaddr.